src/utils/sourcemaps/inject.rs

```diff
@@ -378,6 +378,59 @@ pub fn normalize_sourcemap_url(source_url: &str, sourcemap_url: &str) -> String
     format!("{}{}", &joined[..cutoff], clean_path(&joined[cutoff..]))
 }
 
+/// Returns a list of those paths among `candidate_paths` that differ from `expected_path` in
+/// at most one segment (modulo `.` segments).
+///
+/// If `expected_path` occurs among the `candidate_paths`, no other paths will be returned since
+/// that is considered a unique best match.
+///
+/// The intedend usecase is finding sourcemaps even if they reside in a different directory; see
+/// the `test_find_matching_paths_sourcemaps` test for a minimal example.
+pub fn find_matching_paths(candidate_paths: &[String], expected_path: &str) -> Vec<String> {
+    let mut matches = Vec::new();
+    for candidate in candidate_paths {
+        let candidate_segments = candidate
+            .split('/')
+            .filter(|&segment| segment != ".")
+            .collect::<Vec<_>>();
+        let expected_segments = expected_path
+            .split('/')
+            .filter(|&segment| segment != ".")
+            .collect::<Vec<_>>();
+
+        // If there is a candidate that is exactly equal to the goal path,
+        // return only that one.
+        if candidate_segments == expected_segments {
+            return vec![candidate.clone()];
+        }
+
+        let mut candidate_segments = candidate_segments.into_iter().peekable();
+        let mut expected_segments = expected_segments.into_iter().peekable();
```

Comment:
```suggestion
        let mut candidate_segments = candidate_segments.iter().peekable();
        let mut expected_segments = expected_segments.iter().peekable();
```

There is no need to consume the iter really.